### PR TITLE
feat: get project detail

### PR DESCRIPTION
**Pull request by mkhoi1998**
#### What's this PR do?

- [x] Add get project details by id api

#### Screenshots (if appropriate)

image

### CODE REVIEW

File: pkg/handler/project/project.go [hunk 1]
```diff
@@ -766,3 +766,44 @@ func (h *handler) assignMemberToProject(db *gorm.DB, projectID string, req Assig
...
+	projectID := c.Param("id")
+	if projectID == "" || !model.IsUUIDFromString(projectID) {
+		c.JSON(http.StatusBadRequest, view.CreateResponse[any](nil, nil, ErrInvalidProjectID, nil, ""))
+		return
+	}
+
+	// TODO: can we move this to middleware ?
+	l := h.logger.Fields(logger.Fields{
+		"handler": "project",
+		"method":  "AssignMember",
```
Comment (nguyennh4522) [inline pkg/handler/project/project.go:792]:
```suggestion
		"method":  "Details",
```

File: pkg/handler/project/project_test.go [hunk 2]
```diff
@@ -407,3 +407,55 @@ func TestHandler_AssignMember(t *testing.T) {
 		})
 	}
 }
+
+func TestHandler_GetProject(t *testing.T) {
```
Comment (nguyennh4522) [inline pkg/handler/project/project_test.go:411]:
```suggestion
func TestHandler_Details(t *testing.T) {
```

File: pkg/handler/project/project_test.go [hunk 3]
```diff
@@ -407,3 +407,55 @@ func TestHandler_AssignMember(t *testing.T) {
...
+			h.Details(ctx)
+			require.Equal(t, tt.wantCode, w.Code)
+			expRespRaw, err := ioutil.ReadFile(tt.wantResponsePath)
+			require.NoError(t, err)
+
+			res := w.Body.Bytes()
+			res, _ = utils.RemoveFieldInResponse(res, "createdAt")
+			res, _ = utils.RemoveFieldInResponse(res, "updatedAt")
+
+			require.JSONEq(t, string(expRespRaw), string(res), "[Handler.Project.GetMembers] response mismatched")
```
Comment (nguyennh4522) [inline pkg/handler/project/project_test.go:458]:
```suggestion
			require.JSONEq(t, string(expRespRaw), string(res), "[Handler.Project.Details] response mismatched")
```

File: pkg/handler/project/project.go [hunk 4]
```diff
@@ -766,3 +766,44 @@ func (h *handler) assignMemberToProject(db *gorm.DB, projectID string, req Assig
...
+// Details godoc
+// @Summary Get details of a project
+// @Description Get details of a project
+// @Tags Project
+// @Accept  json
+// @Produce  json
+// @Param Authorization header string true "jwt token"
+// @Param id path string true "Project ID"
+// @Success 200 {object} view.ProjectListDataResponse
+// @Failure 400 {object} view.ErrorResponse
```
Comment (datphamcode295) [inline pkg/handler/project/project.go:779]: add 404 error

File: pkg/handler/project/project.go [hunk 5]
```diff
@@ -766,3 +766,44 @@ func (h *handler) assignMemberToProject(db *gorm.DB, projectID string, req Assig
...
+		"handler": "project",
+		"method":  "AssignMember",
+		"id":      projectID,
+	})
+
+	project, err := h.store.Project.One(h.repo.DB(), projectID)
+	if err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			l.Info("project not found")
+			c.JSON(http.StatusNotFound, view.CreateResponse[any](nil, nil, err, nil, ""))
```
Comment (datphamcode295) [inline pkg/handler/project/project.go:800]: I think it should be more specific like "project not found" not "record not found"

File: pkg/store/project/project.go [hunk 6]
```diff
@@ -78,3 +78,15 @@ func (s *store) Exists(db *gorm.DB, id string) (bool, error) {
 	query := db.Raw("SELECT EXISTS (SELECT * FROM projects WHERE id = ?) as result", id)
 	return record.Result, query.Scan(&record).Error
 }
+
+// One get 1 project by id
+func (s *store) One(db *gorm.DB, id string) (*model.Project, error) {
+	var project *model.Project
+	return project, db.Where("id = ?", id).
+		Preload("Members", "deleted_at IS NULL and left_date IS NULL AND status = ?", model.ProjectMemberStatusActive).
```
Comment (namnhce) [inline pkg/store/project/project.go:86]: Not only `active` member. We have 2 types of `active` member. `active` and `on-boarding`

**Review by namnhce — CHANGES_REQUESTED** [review]

File: pkg/handler/project/testdata/get_project/200.json [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+	"data": {
+		"id": "8dc3be2e-19a4-4942-8a79-56db391a0b15",
+		"createdAt": "",
+		"updatedAt": "",
+		"name": "Fortress",
+		"type": "dwarves",
+		"status": "active",
+		"startDate": "2022-11-01T00:00:00Z",
+		"endDate": null,
+		"members": [
```
Comment (namnhce) [inline pkg/handler/project/testdata/get_project/200.json:11]: missing project stacks in the response

**Review by namnhce — CHANGES_REQUESTED** [review]

File: pkg/handler/project/testdata/get_project/200.json [hunk 8]
```diff
@@ -0,0 +1,40 @@
+{
+	"data": {
+		"id": "8dc3be2e-19a4-4942-8a79-56db391a0b15",
+		"createdAt": "",
+		"updatedAt": "",
+		"name": "Fortress",
+		"type": "dwarves",
+		"status": "active",
+		"startDate": "2022-11-01T00:00:00Z",
+		"endDate": null,
```
Comment (namnhce) [inline pkg/handler/project/testdata/get_project/200.json:10]: missing Client Email, Project email, Country, and Industry fields. Until now, we don't have data and design for them. So leave these fields with TBD value:

```suggestion
		"endDate": null,
		"projectEmail": "TBD",
		"clientEmail": "TBD",
		"industry": "TBD",
		"country": "TBD",
```

**Review by namnhce — CHANGES_REQUESTED** [review]

File: migrations/test_seed/seed.sql [hunk 9]
```diff
@@ -63,3 +63,6 @@ INSERT INTO public.project_slots (id, deleted_at, created_at, updated_at, projec
 INSERT INTO public.project_members (id, deleted_at, created_at, updated_at, project_id, project_slot_id ,employee_id, seniority_id, joined_date, left_date, rate, discount, status,deployment_type, upsell_person_id) VALUES
 ('cb889a9c-b20c-47ee-83b8-44b6d1721aca', null, '2022-11-11 18:19:56.156172', '2022-11-11 18:19:56.156172', '8dc3be2e-19a4-4942-8a79-56db391a0b15', 'f32d08ca-8863-4ab3-8c84-a11849451eb7', '2655832e-f009-4b73-a535-64c3a22e558f', 'dac16ce6-9e5a-4ff3-9ea2-fdea4853925e' ,'2022-11-01', null, 5000, 0, 'active', 'official', '2655832e-f009-4b73-a535-64c3a22e558f'),
 ('7310b51a-3855-498b-99ab-41aa82934269', null, '2022-11-11 18:19:56.156172', '2022-11-11 18:19:56.156172', '8dc3be2e-19a4-4942-8a79-56db391a0b15', 'bdc64b18-4c5f-4025-827a-f5b91d599dc7', 'ecea9d15-05ba-4a4e-9787-54210e3b98ce', 'dac16ce6-9e5a-4ff3-9ea2-fdea4853925e' ,'2022-11-01', null, 4000, 0, 'active', 'shadow', 'ecea9d15-05ba-4a4e-9787-54210e3b98ce');
+
+INSERT INTO public.project_stacks (id, project_id, stack_id) VALUES
+('e7a0dc5e-7da6-48df-879a-f392002732c3', '8dc3be2e-19a4-4942-8a79-56db391a0b15', '0ecf47c8-cca4-4c30-94bb-054b1124c44f');
```
Comment (datphamcode295) [inline migrations/test_seed/seed.sql:68]: add new line the avoid "No newline at end of file" warning on Github

File: pkg/store/project/project.go [hunk 10]
```diff
@@ -78,3 +78,17 @@ func (s *store) Exists(db *gorm.DB, id string) (bool, error) {
...
+// One get 1 project by id
+func (s *store) One(db *gorm.DB, id string) (*model.Project, error) {
+	var project *model.Project
+	return project, db.Where("id = ?", id).
+		Preload("Members", "deleted_at IS NULL and left_date IS NULL AND status IN ?",
+			[]model.AccountStatus{model.AccountStatusActive, model.AccountStatusOnBoarding}).
+		Preload("Members.Employee").
+		Preload("Heads", "deleted_at IS NULL and left_date IS NULL").
+		Preload("Heads.Employee").
+		Preload("Stacks").
```
Comment (datphamcode295) [inline pkg/store/project/project.go:91]: I'm not sure "Stacks" in project is the same with in employee. If it's similar, it should be check table project_stacks with condition deleted_at is null before preload stacks. @namnhce help me, please

File: pkg/view/stack.go [hunk 11]
```diff
@@ -7,7 +7,7 @@ type Stack struct {
 	Code string `json:"code"`
 }
 
-func ToStacks(stacks []model.EmployeeStack) []Stack {
+func ToEmployeeStacks(stacks []model.EmployeeStack) []Stack {
```
Comment (namnhce) [inline pkg/view/stack.go:10]: 2 hàm chọn 1 hàm thôi em @mkhoi1998

**Review by namnhce — CHANGES_REQUESTED** [review]

**Review by namnhce — APPROVED** [review]

**Pull request merged**